src/coroutine/cancel.rs

```diff
@@ -1,16 +1,16 @@
 use crate::{ffi, Py, PyAny, PyObject};
-use futures_util::future::poll_fn;
-use futures_util::task::AtomicWaker;
+use std::future::Future;
+use std::pin::Pin;
 use std::ptr;
 use std::ptr::NonNull;
 use std::sync::atomic::{AtomicPtr, Ordering};
-use std::sync::Arc;
-use std::task::{Context, Poll};
+use std::sync::{Arc, Mutex};
```

Comment:
I think we use `parking_lot::Mutex` for our other internal mutex (in the `gil` module). I think it would improve consistency to use it here as well.

Reply:
FWIW, I've been sitting on a patch that drops parking_lot for our existing uses, by raising the MSRV to 1.60.

I think it's probably either practical now or will be very soon. (For pyca/cryptography, our next release will go to an MSRV of 1.63)